private/buf/bufgen/generator.go

```diff
@@ -322,6 +327,52 @@ func (g *generator) execRemotePlugin(
 	return responses[0], nil
 }
 
+func (g *generator) execRemotePluginV2(
+	ctx context.Context,
+	container app.EnvStdioContainer,
+	image bufimage.Image,
+	pluginConfig *PluginConfig,
+	includeImports bool,
+	includeWellKnownTypes bool,
+) (*pluginpb.CodeGeneratorResponse, error) {
+	reference, err := bufpluginref.PluginReferenceForString(pluginConfig.Plugin)
+	if err != nil {
+		return nil, fmt.Errorf("invalid plugin path: %w", err)
+	}
+	codeGenerationService, err := g.registryProvider.NewCodeGenerationService(ctx, reference.Remote())
+	if err != nil {
+		return nil, fmt.Errorf("failed to create code generation service for remote %q: %w", reference.Remote(), err)
+	}
+	var options []string
+	if len(pluginConfig.Opt) > 0 {
+		// Only include parameters if they're not empty.
+		options = []string{pluginConfig.Opt}
+	}
+	responses, err := codeGenerationService.GenerateCode(
+		ctx,
+		bufimage.ImageToProtoImage(image),
+		[]*registryv1alpha1.PluginGenerationRequest{
+			{
+				PluginReference: bufplugin.PluginReferenceToProtoCuratedPluginReference(reference),
+				Options:         options,
+			},
+		},
```

Comment:
It looks like we're still just planning to send one plugin at a time (but concurrently), so we might not need the repeated field to begin with after all. But this is at least symmetrical with the old model, so it seems fine to leave it as-is.

Reply:
I didn't look at this in detail before but if this is the expected calling pattern do you think we should change the API to match how it is used? Do we think there will be worthwhile optimizations of executing multiple plugins in batch on the backend?

Reply:
Technically updating a single `string` field to be `repeated` is a backwards wire compatible change ([link](https://developers.google.com/protocol-buffers/docs/proto#updating)), so we should be able to make this change later on if ever. The only thing is if we change the field name - starting with a single `string` would suggest that it should be called a `request` (rather than `requests`), which will be another form of code-level incompatibility.

Batching them would prevent multiple round trips which is nice. Plus, it doesn't really do us any harm to support this feature out the gate, so I'm OK with keeping this as `repeated` for now. We'll be the only caller for the foreseeable future, so we can always change it later on before the BSR API is actually consumed by others.